action/protocol/execution/protocol_test.go

```diff
@@ -505,7 +504,8 @@ func TestProtocol_Handle(t *testing.T) {
 		require.Equal(0, balance.Cmp(big.NewInt(500000000)))
 
 		log.S().Info("Roll Dice")
-		data, _ = hex.DecodeString("797d9fbd000000000000000000000000d950673630e2286adc157c35f2fd73a1ef49d40e")
+		h := keypair.HashPubKey(testaddress.Keyinfo["alfa"].PubKey)
```

Comment:
we need to change the HashPubKey() implementation, according to https://kobl.one/blog/create-full-ethereum-keypair-and-address/
1. digest = keccak-256(hexdecimal format of the pubkey, not bytes)
2. ethereum address = digest[12:]
i am checking ethereum code to confirm

Reply:
good catch